src/libraries/Microsoft.PowerFx.Core/Public/Values/CustomObjectValue.cs

```diff
@@ -11,17 +11,19 @@
 
 namespace Microsoft.PowerFx.Core.Public.Values
 {
-    public interface ICustomObject
+    public enum CustomObjectKind
```

Comment:
>CustomObjectKind [](http://example.com/codeflow?start=16&length=16)

Some notes for design discussion:

1. How will it handle future ICustomObject impls beyond json. Like Xml, which has more date,time and other types.

2. The thunk idea where we create a ICustomObj impl around a FormulaValue. Or wrap around a C# object (ala a dynamic version of FormulaValue.New)

3. Potential overlap with other FormulasValues. Eg,  ICustomObject.TryGetProperty() vs RecordValue.GetField(). Maybe there are some common interfaces?

The sole purpose of an ICustomObject is to produce FormulaValues, consider if ICustomObject.Kind should return a FormulaType rather than a new enum (which then needs to just be converted to a formula type).

Reply:
We discussed offline - 
this needs to describe foriegn types (like json arrays), so we'll need to keep the enum - we can't change it to FormulaType.